### PR TITLE
Fix logging filename lookup.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

- runtime.Callers may be inlined so don't try to skip it.
- Increase callers buffer, new versions of logrus use a deeper stack.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Fix that Felix logs would sometimes record <nil> for the filename of some logs.
```

### CODE REVIEW

**Review by neiljerram — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
